Fixes #8934 - Replace Firefox-specific passcode with native device authentication

st3fan opened the pull request:
This patch removes the Firefox-specific passcode and replaces it with the system provided _LocalAuthentication_ instead.

The following changes were made:

 - Remove all code related to the Firefox-specific passcode
 - TODO Add a view to explain you cannot access your logins if you have not set a device passcode.
 - TODO

st3fan commented:
Even though the string _may_ change a bit, I am already asking for a code review. Trying to get this landed this week so that it has a high chance of being part of Firefox 39.

st3fan commented:
@brampitoyo  I addressed all your suggestions.

st3fan commented:
Note to self, remove the `AuthenticationKeychainInfoTests` test. Also interesting my PR was green for weeks. Are we now running more tests? Let’s run the full suite locally to see what else fails or needs to be fixed.

st3fan commented:
@brampitoyo This is the latest. I put in the revised text by Brian and followed your advice to change the button to _Cancel_.

![Screen Shot 2021-10-16 at 9 34 44 AM](https://user-images.githubusercontent.com/28052/137798856-7f585069-548b-4797-9776-af9eb828feb6.png)

nbhasin2 commented:
@st3fan the PR has the following items we can improve and refactor, let me know if you'd like to do it here or in a follow-up PR and I am good to approve these changes

- Snapkit 
- View initialization refactor and remove items from viewDidLoad

st3fan commented:
@nbhasin2 thank you very much for the thorough review 💯 

 - I moved all the subview setup in lazy variables, which makes the `viewDidLoad` a lot more concise 👍🏻  Both for 
 `DevicePasscodeRequiredViewController` and `LoginOnboardingViewController`.
- Removed unused strings
- Fixed a small bug where the top padding of the _Learn More_ link was `10` in one view and `20` in the other
- Filed #9378 for the SnapKit removal 🥳 which I'll submit before code freeze for v40
